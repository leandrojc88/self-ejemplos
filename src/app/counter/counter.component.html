<h1>Counter {{ counter }}</h1>
<hr />

<button class="btn btn-primary me-2" (click)="setCounter(1)">+1</button>
<button class="btn btn-primary" (click)="setCounter(-1)">-1</button>
