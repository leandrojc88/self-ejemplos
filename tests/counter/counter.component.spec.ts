import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../src/app/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CounterComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('shoild be equal to snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('should by setCounter(5) to 5', () => {
    component.setCounter(5);
    expect(component.counter).toBe(5);
  });

  it('should by on click to button +1 and button -1', () => {
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    buttons[0].click();
    buttons[0].click();
    expect(component.counter).toBe(3);
    buttons[1].click();
    expect(component.counter).toBe(2);
  });

  it('should incresad counter and view in html', () => {
    component.setCounter(5);
    fixture.detectChanges();

    const h1 = compiled.querySelector('h1');
    expect(h1?.textContent).toContain('5');
  });
});
