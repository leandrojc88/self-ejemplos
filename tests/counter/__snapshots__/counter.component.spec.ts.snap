// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent shoild be equal to snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 0
  </h1>
  <hr />
  <button
    class="btn btn-primary me-2"
  >
    +1
  </button>
  <button
    class="btn btn-primary"
  >
    -1
  </button>
</div>
`;
